### ISSUE TITLE
pmi_client (v1) test fails at larger scales

### ISSUE THREAD

**Issue by garlick** [pmix/master#85]
The `pmi_client` test, when run at sizes greater than about 10 fails consistently for me:
```
$ ./pmix_test  -n 16 -e ./pmi_client 
ERROR [cli_stages.c:252:errhandler]:  PMIX server Error handler with status = -27
ERROR [cli_stages.c:148:cli_cleanup]: rank 0 with state 1 unexpectedly terminated.
ERROR [cli_stages.c:252:errhandler]:  PMIX server Error handler with status = -27
ERROR [cli_stages.c:252:errhandler]:  PMIX server Error handler with status = -27
ERROR [pmix_test.c:167:main]: Test was aborted!
```

Here's a run with some debug
```
$ VERBOSE=3 ./pmix_test -v  -n 16 -e ./pmi_client 
pmix_test.c:main: Start PMIx_lite smoke test (timeout is 10)
pmix_test.c:main: Executing test: ./pmi_client -n 16 -v
utils.c:launch_clients: Setting job info
cli_stages.c:errhandler_reg_callbk: ERRHANDLER REGISTRATION CALLBACK CALLED WITH STATUS 0, ref=0
INFO  spawned=0 size=16 rank=3 appnum=22872
INFO  PMI_Get_id_length_max=255
INFO  jobid=smoky_nspace-0
INFO  PMI_Get_kvs_domain_id=smoky_nspace-0
INFO  PMI_KVS_Get_my_name=smoky_nspace-0
INFO  TI1  : PASS
INFO  TI2  : PASS
INFO  PMI_KVS_Get_key_length_max=511
INFO  PMI_KVS_Get_value_length_max=4096
INFO  TI3  : PASS
INFO  PMI_Get_clique_size=16
INFO  spawned=0 size=16 rank=2 appnum=22872
INFO  PMI_Get_id_length_max=255
INFO  jobid=smoky_nspace-0
INFO  PMI_Get_kvs_domain_id=smoky_nspace-0
INFO  PMI_KVS_Get_my_name=smoky_nspace-0
INFO  TI1  : PASS
INFO  TI2  : PASS
INFO  PMI_KVS_Get_key_length_max=511
INFO  PMI_KVS_Get_value_length_max=4096
INFO  TI3  : PASS
INFO  PMI_Get_clique_size=16
INFO  spawned=0 size=16 rank=1 appnum=22872
INFO  PMI_Get_id_length_max=255
INFO  jobid=smoky_nspace-0
INFO  PMI_Get_kvs_domain_id=smoky_nspace-0
INFO  PMI_KVS_Get_my_name=smoky_nspace-0
INFO  TI1  : PASS
INFO  TI2  : PASS
INFO  PMI_KVS_Get_key_length_max=511
INFO  PMI_KVS_Get_value_length_max=4096
INFO  TI3  : PASS
INFO  PMI_Get_clique_size=16
INFO  spawned=0 size=16 rank=7 appnum=22872
INFO  PMI_Get_id_length_max=255
INFO  jobid=smoky_nspace-0
INFO  PMI_Get_kvs_domain_id=smoky_nspace-0
INFO  PMI_KVS_Get_my_name=smoky_nspace-0
INFO  TI1  : PASS
INFO  TI2  : PASS
INFO  PMI_KVS_Get_key_length_max=511
INFO  PMI_KVS_Get_value_length_max=4096
INFO  TI3  : PASS
INFO  PMI_Get_clique_size=16
INFO  spawned=0 size=16 rank=8 appnum=22872
INFO  PMI_Get_id_length_max=255
INFO  jobid=smoky_nspace-0
INFO  PMI_Get_kvs_domain_id=smoky_nspace-0
INFO  PMI_KVS_Get_my_name=smoky_nspace-0
INFO  TI1  : PASS
INFO  TI2  : PASS
INFO  PMI_KVS_Get_key_length_max=511
INFO  PMI_KVS_Get_value_length_max=4096
INFO  TI3  : PASS
INFO  PMI_Get_clique_size=16
INFO  spawned=0 size=16 rank=12 appnum=22872
INFO  spawned=0 size=16 rank=4 appnum=22872
INFO  PMI_Get_id_length_max=255
INFO  jobid=smoky_nspace-0
INFO  PMI_Get_kvs_domain_id=smoky_nspace-0
INFO  PMI_KVS_Get_my_name=smoky_nspace-0
INFO  TI1  : PASS
INFO  TI2  : PASS
INFO  PMI_KVS_Get_key_length_max=511
INFO  PMI_KVS_Get_value_length_max=4096
INFO  TI3  : PASS
INFO  spawned=0 size=16 rank=14 appnum=22872
INFO  PMI_Get_id_length_max=255
INFO  spawned=0 size=16 rank=9 appnum=22872
INFO  PMI_Get_id_length_max=255
INFO  jobid=smoky_nspace-0
INFO  PMI_Get_kvs_domain_id=smoky_nspace-0
INFO  PMI_KVS_Get_my_name=smoky_nspace-0
INFO  TI1  : PASS
INFO  TI2  : PASS
INFO  PMI_KVS_Get_key_length_max=511
INFO  PMI_KVS_Get_value_length_max=4096
INFO  TI3  : PASS
INFO  jobid=smoky_nspace-0
INFO  PMI_Get_kvs_domain_id=smoky_nspace-0
INFO  PMI_KVS_Get_my_name=smoky_nspace-0
INFO  PMI_Get_id_length_max=255
INFO  jobid=smoky_nspace-0
INFO  PMI_Get_kvs_domain_id=smoky_nspace-0
INFO  PMI_KVS_Get_my_name=smoky_nspace-0
INFO  TI1  : PASS
INFO  TI2  : PASS
INFO  PMI_KVS_Get_key_length_max=511
INFO  PMI_KVS_Get_value_length_max=4096
INFO  TI3  : PASS
INFO  TI1  : PASS
INFO  spawned=0 size=16 rank=15 appnum=22872
INFO  TI2  : PASS
INFO  PMI_KVS_Get_key_length_max=511
INFO  PMI_KVS_Get_value_length_max=4096
INFO  PMI_Get_clique_size=16
INFO  TI3  : PASS
INFO  PMI_Get_clique_size=16
INFO  spawned=0 size=16 rank=0 appnum=22872
INFO  PMI_Get_id_length_max=255
INFO  jobid=smoky_nspace-0
INFO  PMI_Get_kvs_domain_id=smoky_nspace-0
INFO  PMI_KVS_Get_my_name=smoky_nspace-0
INFO  TI1  : PASS
INFO  TI2  : PASS
INFO  PMI_KVS_Get_key_length_max=511
INFO  PMI_KVS_Get_value_length_max=4096
INFO  TI3  : PASS
INFO  PMI_Get_clique_size=16
INFO  PMI_Get_id_length_max=255
INFO  PMI_Get_clique_size=16
INFO  jobid=smoky_nspace-0
INFO  PMI_Get_kvs_domain_id=smoky_nspace-0
INFO  PMI_KVS_Get_my_name=smoky_nspace-0
INFO  TI1  : PASS
INFO  TI2  : PASS
INFO  PMI_KVS_Get_key_length_max=511
INFO  PMI_KVS_Get_value_length_max=4096
INFO  TI3  : PASS
INFO  PMI_Get_clique_size=16
INFO  PMI_Get_clique_size=16
INFO  spawned=0 size=16 rank=13 appnum=22872
INFO  spawned=0 size=16 rank=6 appnum=22872
INFO  PMI_Get_id_length_max=255
INFO  jobid=smoky_nspace-0
INFO  PMI_Get_kvs_domain_id=smoky_nspace-0
INFO  PMI_KVS_Get_my_name=smoky_nspace-0
INFO  TI1  : PASS
INFO  TI2  : PASS
INFO  PMI_KVS_Get_key_length_max=511
INFO  PMI_KVS_Get_value_length_max=4096
INFO  TI3  : PASS
INFO  PMI_Get_id_length_max=255
INFO  PMI_Get_clique_size=16
INFO  jobid=smoky_nspace-0
INFO  PMI_Get_kvs_domain_id=smoky_nspace-0
INFO  PMI_KVS_Get_my_name=smoky_nspace-0
INFO  TI1  : PASS
INFO  TI2  : PASS
INFO  PMI_KVS_Get_key_length_max=511
INFO  PMI_KVS_Get_value_length_max=4096
INFO  TI3  : PASS
INFO  PMI_Get_clique_size=16
INFO  spawned=0 size=16 rank=5 appnum=22872
INFO  PMI_Get_id_length_max=255
INFO  jobid=smoky_nspace-0
INFO  PMI_Get_kvs_domain_id=smoky_nspace-0
INFO  PMI_KVS_Get_my_name=smoky_nspace-0
INFO  TI1  : PASS
INFO  TI2  : PASS
INFO  PMI_KVS_Get_key_length_max=511
INFO  PMI_KVS_Get_value_length_max=4096
INFO  TI3  : PASS
INFO  spawned=0 size=16 rank=11 appnum=22872
INFO  PMI_Get_clique_size=16
INFO  PMI_Get_id_length_max=255
INFO  jobid=smoky_nspace-0
INFO  PMI_Get_kvs_domain_id=smoky_nspace-0
INFO  PMI_KVS_Get_my_name=smoky_nspace-0
INFO  TI1  : PASS
INFO  TI2  : PASS
INFO  PMI_KVS_Get_key_length_max=511
INFO  PMI_KVS_Get_value_length_max=4096
INFO  TI3  : PASS
INFO  spawned=0 size=16 rank=10 appnum=22872
INFO  PMI_Get_id_length_max=255
INFO  jobid=smoky_nspace-0
INFO  PMI_Get_kvs_domain_id=smoky_nspace-0
INFO  PMI_Get_clique_size=16
INFO  PMI_KVS_Get_my_name=smoky_nspace-0
INFO  TI1  : PASS
INFO  TI2  : PASS
INFO  PMI_KVS_Get_key_length_max=511
INFO  PMI_KVS_Get_value_length_max=4096
INFO  TI3  : PASS
INFO  PMI_Get_clique_size=16
ERROR [cli_stages.c:252:errhandler]:  PMIX server Error handler with status = -27
cli_stages.c:op_callbk: OP CALLBACK CALLED WITH STATUS 0
cli_stages.c:cli_wait_all: waitpid = 22891
cli_stages.c:cli_wait_all: the child with pid = 22891 has rank = 3
		exited = 0, signalled = 1
ERROR [cli_stages.c:148:cli_cleanup]: rank 3 with state 1 unexpectedly terminated.
cli_stages.c:cli_terminate: Client rank = 3 terminated
ERROR [cli_stages.c:252:errhandler]:  PMIX server Error handler with status = -27
cli_stages.c:op_callbk: OP CALLBACK CALLED WITH STATUS 0
ERROR [pmix_test.c:167:main]: Test was aborted!
cli_stages.c:cli_kill_all: Kill rank 0 (pid = 22888).
cli_stages.c:cli_terminate: Client rank = 0 terminated
cli_stages.c:cli_kill_all: Kill rank 1 (pid = 22889).
cli_stages.c:cli_terminate: Client rank = 1 terminated
cli_stages.c:cli_kill_all: Kill rank 2 (pid = 22890).
cli_stages.c:cli_terminate: Client rank = 2 terminated
cli_stages.c:cli_kill_all: Skip rank 3 as it was already terminated.
cli_stages.c:cli_kill_all: Kill rank 4 (pid = 22892).
cli_stages.c:cli_terminate: Client rank = 4 terminated
cli_stages.c:cli_kill_all: Kill rank 5 (pid = 22893).
cli_stages.c:cli_terminate: Client rank = 5 terminated
cli_stages.c:cli_kill_all: Kill rank 6 (pid = 22894).
cli_stages.c:cli_terminate: Client rank = 6 terminated
cli_stages.c:cli_kill_all: Kill rank 7 (pid = 22895).
cli_stages.c:cli_terminate: Client rank = 7 terminated
cli_stages.c:cli_kill_all: Kill rank 8 (pid = 22898).
cli_stages.c:cli_terminate: Client rank = 8 terminated
cli_stages.c:cli_kill_all: Kill rank 9 (pid = 22902).
cli_stages.c:cli_terminate: Client rank = 9 terminated
cli_stages.c:cli_kill_all: Kill rank 10 (pid = 22904).
cli_stages.c:cli_terminate: Client rank = 10 terminated
cli_stages.c:cli_kill_all: Kill rank 11 (pid = 22906).
cli_stages.c:cli_terminate: Client rank = 11 terminated
cli_stages.c:cli_kill_all: Kill rank 12 (pid = 22909).
cli_stages.c:cli_terminate: Client rank = 12 terminated
cli_stages.c:cli_kill_all: Kill rank 13 (pid = 22911).
cli_stages.c:cli_terminate: Client rank = 13 terminated
cli_stages.c:cli_kill_all: Kill rank 14 (pid = 22914).
cli_stages.c:cli_terminate: Client rank = 14 terminated
cli_stages.c:cli_kill_all: Kill rank 15 (pid = 22916).
cli_stages.c:cli_terminate: Client rank = 15 terminated
cli_stages.c:op_callbk: OP CALLBACK CALLED WITH STATUS 0
```

**Comment by rhc54** [comment]
Took me awhile to track this down, but it was indeed a bug in the test and not the code base itself.

Thanks for point it out. Fixed in pmix/master@bde4da1545072c943cc9be71fc14a10609deef4a

**Comment by garlick** [comment]
Thanks for tracking this one down!
